docs/topics/coroutines.rst

```diff
@@ -0,0 +1,29 @@
+==========
+Coroutines
+==========
+
+.. versionadded:: 2.0
+
+Scrapy has partial support for the :ref:`coroutine syntax <async>`.
+
+The following callables may be defined as coroutines using ``async def``, and
+hence use coroutine syntax (e.g. ``await``, ``async for``, ``async with``):
+
+-   :class:`~scrapy.http.Request` callbacks.
+
+    However, you can only use ``yield`` if you are using Python 3.6 or later.
```

Comment:
Is this a Scrapy limitation of a Python limitation?

Reply:
As @Gallaecio already answered elsewhere, `yield` in an `async def` coroutine makes it an async generator and is supported only since Python 3.6.

Reply:
Maybe we should make that explicit there, or at least link to a resource that explains that?

Reply:
Yeah.

https://docs.python.org/3/glossary.html#term-asynchronous-generator doesn't tell the version reqs :-/

Reply:
We also need to mention here the limitations with the async generator callback processing from #4318